### PR TITLE
Put adaptor state configuration in a struct to avoid leaking data to sentry

### LINKED ISSUE #1600: Consider how Sentry filtering works for errors that may contain credentials
If a superuser enables Sentry error reporting they could report credential secrets via the error objects.

An example of this can be found here: https://openfn.sentry.io/issues/4463772292/?alert_rule_id=11218001&alert_type=issue&environment=prod&notification_uuid=e2fb2c20-6e7a-4064-89bc-7a8a51378641&project=6446500

While we can't omit given arguments for all `CaseClauseError` errors, we should still potentially expand protocol-level obfuscation... @midigofrank suggests rules for `inpsect()` around the `Credential` struct.

### PR DESCRIPTION
**Pull request by midigofrank**
## Notes for the reviewer

- Wraps credential body in a `Lightning.AdaptorState` struct and implements the `Inspect` protocol to hide the data

## Related issue

Fixes #1600 

## Review checklist

- [x] I have performed a **self-review** of my code
- [x] I have verified that all appropriate **authorization policies** have been implemented and tested
- [x] If needed, I have updated the **changelog**
- [ ] Product has **QA'd** this feature

### CODE REVIEW

**Review by taylordowns2000 — COMMENTED** [review]
@midigofrank , are there any other places in the application where we use a credential body? (like the attempts service, state assembler, or even credential edit/view?) would they need to use this new struct too?

**Review by taylordowns2000 — CHANGES_REQUESTED** [review]
per that earlier call, please change `AdaptorState` to `SafeSecrets`

**Review by taylordowns2000 — APPROVED** [review]
looks good. waiting for @stuartc to weigh in here

File: lib/lightning/safe_secrets.ex [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  @type t :: %__MODULE__{configuration: map()}
+  @derive Jason.Encoder
+  defstruct [:configuration]
+
+  defimpl Inspect, for: Lightning.SafeSecrets do
+    def inspect(%{configuration: _config}, _opts) do
+      Kernel.inspect(%{"configuration" => "[FILTERED]"})
+    end
+  end
```
[the remark this answers is not in the document]
Comment (taylordowns2000) [reply lib/lightning/safe_secrets.ex:12]: Great catch @stuartc . I think we actually discussed this already (requested change from `AdaptorState` to `SafeSecrets`) and I missed that it still had the "configuration" bit in my last review.

[the remark this answers is not in the document]
Comment (midigofrank) [reply lib/lightning/safe_secrets.ex:12]: Hmm, makes sense,  let me update it. And nope, there's no issue with nested data structures

**Review by stuartc — APPROVED** [review]

**Review by stuartc — APPROVED** [review]

**Pull request merged**